File: python/src/word_count.py
import re

def clean(sentence):
    return re.sub(
        r"('*[^a-zA-Z0-9']+'*|^'*|'*$)", 
        " ", 
        sentence.lower()
    )

def get_words(sentence):
    return sentence.split()

def count_words(sentence):
    words = get_words(clean(sentence))

    result = {}
    for word in words:
        result[word] = result.get(word, 0) + 1
    return result

File: scala/src/main/scala/WordCount.scala
case class WordCount(words: String) {
  def countWords: Map[String, Int] = {
    words.trim
      .toLowerCase
      .split("'*[^a-z0-9']+'*")
      .filter(s => !s.isBlank)
      .foldLeft(Map[String, Int]()) {
        (map, word) => map + (word -> (map.getOrElse(word, 0) + 1))
      }
  }
}